server/src/main/scala/org/http4s/server/middleware/authentication/Nonce.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.server.middleware.authentication
+
+import org.http4s.crypto.unsafe.SecureRandom
+
+import java.math.BigInteger
+import java.util.Date
```

Comment:
Is this better than a `java.time.Instant`?  That's what we use most places.  It's not public API, so not a huge deal.

Reply:
Swapping this out to `Instant` also let me swap out for `Clock[F].instantNow`, good call